### ISSUE TITLE
Recipies: When building with multiple base resources, ETA is wrong

### ISSUE THREAD

**Issue by jd284** [taniwha/Extraplanetary-Launchpads#124]
When using recipies to change the construction materials to require multiple materials in different amounts (e.g. like MKS does with requiring 80% MaterialKits and 20% SpecializedParts by mass), the displayed ETA is wrong, and with it the automatic KAC Alarm.

What happens is that initially, both resources have their individual ETA doubled since they are being built in parallel. That's fine. However the amount of resources is different, so after one resource is finished, the other has its time suddenly cut in half from the original estimate because now there's only one resource left to build.
So either the initial estimate should take this progression into account (basically the longest time should be clamped to the total KerbalHours), or all the resources should complete at the same time, in other words the smaller resources should finish more slowly so that all of them take exactly the total KerbalHours.

Example:
Constructing a 6 t rocket requires 4800 MaterialKits and 317.5 SpecializedParts.
For a productivity of 1, the ETA says it will take 4h48m for the MaterialKits and 1h12m for the SpecializedParts.
Just before the 1h12m has passed and the SpecializedParts have been completed, there is 3h36m left on the MaterialKits.
However, immediately afterwards this number jumps to 1h48m (half the previous time). After this 1h48m, the construction is done.
In other words, the original estimate of the MaterialKits ETA was 1h48m too long. Instead it should've been the 3h you get from 6t * 0.5 KerbalHours/t, which is how long it really took (1h12m + 1h48m = 3h).

This is mainly annoying when using the KAC Alarms, since ship construction will have been done long before the alarm goes off, unless you manually reset the alarm after SpecializedParts have completed.

**Comment by taniwha** [comment]
ETA will never be fully accurate (the E is for "estimated", thus by definition it is a guess), but the calculations have been much improved in 6.3.3.
